WordPress/Classes/ViewRelated/Me/App Settings/Boolean User Defaults/BooleanUserDefaultsDebugViewModel.swift

```diff
@@ -0,0 +1,94 @@
+import SwiftUI
+
+final class BooleanUserDefaultsDebugViewModel: ObservableObject {
+    @Published private var allUserDefaultsSections = BooleanUserDefaultsSections()
+    @Published var searchQuery: String = ""
+
+    private var persistentRepository: UserPersistentRepository
+
+    var userDefaultsSections: BooleanUserDefaultsSections {
+        return if searchQuery.isEmpty {
+            allUserDefaultsSections
+        } else {
+            filterUserDefaults(by: searchQuery)
+        }
+    }
+
+    let coreDataStack: CoreDataStack
+
+    private func filterUserDefaults(by query: String) -> BooleanUserDefaultsSections {
+        var filteredSections = BooleanUserDefaultsSections()
+        allUserDefaultsSections.forEach { sectionKey, userDefaults in
+            let filteredUserDefaults = userDefaults.filter { key, userDefault in
+                key.localizedCaseInsensitiveContains(query) || userDefault.title.localizedCaseInsensitiveContains(query)
+            }
+            if sectionKey.localizedCaseInsensitiveContains(query) || !filteredUserDefaults.isEmpty {
+                filteredSections[sectionKey] = filteredUserDefaults.isEmpty ? userDefaults : filteredUserDefaults
+            }
+        }
+        return filteredSections
+    }
+
+    init() {
+        persistentRepository = UserPersistentStoreFactory.instance()
+        coreDataStack = ContextManager.shared
+        load()
+    }
+
+    func load() {
+        let allUserDefaults = persistentRepository.dictionaryRepresentation()
+        var loadedUserDefaultsSections = BooleanUserDefaultsSections()
+
+        for (entryKey, entryValue) in allUserDefaults {
+            if let groupedUserDefaults = entryValue as? [String: Bool] {
+                loadedUserDefaultsSections[entryKey] = processGroupedUserDefaults(groupedUserDefaults)
+            } else if let booleanUserDefault = entryValue as? Bool, !isSystemUserDefault(entryKey) {
+                loadedUserDefaultsSections[Strings.otherBooleanUserDefaultsSectionID, default: [:]][entryKey] = BooleanUserDefault(title: entryKey, value: booleanUserDefault)
```

Comment:
[Suggestion] This line can be simplified to make it more readable by creating an `others` section as a variable outside of the loop. Set the values here, and assign the whole section after the loop. Something like this:
```swift
    func load() {
        let allUserDefaults = persistentRepository.dictionaryRepresentation()
        var loadedUserDefaultsSections = BooleanUserDefaultsSections()
        var otherSection = BooleanUserDefaults()
        for (entryKey, entryValue) in allUserDefaults {
            if let groupedUserDefaults = entryValue as? [String: Bool] {
                loadedUserDefaultsSections[entryKey] = processGroupedUserDefaults(groupedUserDefaults)
            } else if let booleanUserDefault = entryValue as? Bool, !isSystemUserDefault(entryKey) {
                otherSection[entryKey] = BooleanUserDefault(title: entryKey, value: booleanUserDefault)
            }
        }
        loadedUserDefaultsSections[Strings.otherBooleanUserDefaultsSectionID] = otherSection
        allUserDefaultsSections = loadedUserDefaultsSections
    }
```